src/DoctrineBatchUtils/BatchProcessing/SelectBatchIteratorAggregate.php

```diff
@@ -20,32 +19,49 @@
  * 'Read' focused batching iterator that will issue a clear to the entity manager
  * every batch size to 'detach' the managed objects and therefore make them entities only for the
  * purpose of reading without manually attaching them back to the EntityManager.
+ *
+ * @template A
+ * @template B
  */
 final class SelectBatchIteratorAggregate implements IteratorAggregate
 {
-    /** @var Traversable<mixed> */
-    private Traversable $resultSet;
+    /** @var iterable<A, B> */
+    private iterable $resultSet;
     private EntityManagerInterface $entityManager;
+    /** @psalm-var positive-int */
     private int $batchSize;
 
+    /**
+     * @psalm-param positive-int $batchSize
+     */
     public static function fromQuery(AbstractQuery $query, int $batchSize): self
     {
-        return new self($query->iterate(), $query->getEntityManager(), $batchSize);
+        return new self($query->toIterable(), $query->getEntityManager(), $batchSize);
```

Comment:
I think this requires raising minimum ORM version dependency

Reply:
doctrine/orm is already ^2.8. see : https://github.com/Ocramius/DoctrineBatchUtils/blob/8affefde661133a385c2416dcb63277e77e1913b/composer.json#L22